components/console/src/core/infrastructure/mongo/repositories/mongo-organization-avatar-repository.ts

```diff
@@ -0,0 +1,150 @@
+  public get model(): typeof OrganizationAvatar {
+    return OrganizationAvatar
+  }
+
+  /**
+   * Creates a new organization avatar document in MongoDB.
+   * @param organizationAvatar - The organization avatar entity to persist.
+   * @returns A promise that resolves to the persisted organization avatar entity.
+   * @throws {DatabaseException} If the database operation fails.
+   */
+  async create(
+    organizationAvatar: OrganizationAvatarEntity
+  ): Promise<OrganizationAvatarEntity> {
+    try {
+      const result = await this.model.create(organizationAvatar)
+      const OrganizationAvatarEntity = OrganizationAvatarMapper.toEntity(result)
+
+      return OrganizationAvatarEntity
+    } catch (error) {
+      this.logger.error('[ERROR] - MongoOrganizationAvatarRepository.create', {
+        error,
+        context: 'mongo'
+      })
+
+      throw handleDatabaseError(error)
+    }
+  }
+
+  /**
+   * Updates an existing organization avatar document in MongoDB.
+   * @param organizationAvatar - The organization avatar entity to update.
+   * @returns A promise that resolves to the updated organization avatar entity.
+   * @throws {DatabaseException} If the database operation fails.
+   */
+  async update(
+    organizationAvatar: OrganizationAvatarEntity
+  ): Promise<OrganizationAvatarEntity> {
+    try {
+      await this.model.updateOne(
+        { organizationId: organizationAvatar.organizationId },
+        organizationAvatar
+      )
+
+      return organizationAvatar as OrganizationAvatarEntity
+    } catch (error) {
+      this.logger.error('[ERROR] - MongoOrganizationAvatarRepository.update', {
+        error,
+        context: 'mongo'
+      })
+
+      throw handleDatabaseError(error)
+    }
+  }
+
+  /**
+   * Deletes the organization avatar document by organization ID.
+   * @param organizationId - The ID of the organization whose avatar to delete.
+   * @returns A Promise that resolves when deletion is complete.
+   * @throws {DatabaseException} If the database operation fails.
+   */
+  async delete(organizationId: string): Promise<void> {
+    try {
+      await this.model.deleteOne({
+        organizationId: organizationId
+      })
+    } catch (error) {
+      this.logger.error('[ERROR] - MongoOrganizationAvatarRepository.delete', {
+        error,
+        context: 'mongo'
+      })
+
+      throw handleDatabaseError(error)
+    }
+  }
+
+  /**
+   * Retrieves the organization avatar by organization ID.
+   * @param organizationId - The ID of the organization.
+   * @returns A promise that resolves to the found organization avatar entity, or undefined if none exists.
+   * @throws {DatabaseException} If the database operation fails.
+   */
+  async fetchById(
+    organizationId: string
+  ): Promise<OrganizationAvatarEntity | undefined> {
+    try {
+      const result = await this.model
+        .findOne({
+          organizationId
+        })
+        .lean()
+
+      const organizationAvatarEntity = result
```

Comment:
We should throw an exception if "fetchById" fails.
Throw the NotFoundApiException from @/lib/http here.